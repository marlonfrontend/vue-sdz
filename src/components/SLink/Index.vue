<template>
  <component
    :is="type"
    :to="to || null"
    :href="external && to"
    :target="external && '_blank'"

    v-bind="$attrs"

    class="c-link"

    v-on="$listeners"
  >
    <!-- @slot The text/label of the link -->
    <slot />
  </component>
</template>

<script>

/**
 * Generic link component.
 */
export default {
  name: 'SLink',

  props: {
    /**
     * The link's path, can be either an external URL or the path to an
     * internal route in the app (must follow the vue-router api).
     */
    to: {
      type: [String, Object],
      default: () => null
    },

    link: Boolean,

    /**
     * Whether the link is external or not
     */
    external: {
      type: Boolean,
      default: false
    },

    /**
     * Whether to use the nuxt component for route links.
     */
    nuxt: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    type () {
      if (this.link) return 'a'

      return !this.to
        ? 'button' : this.external
          ? 'a' : this.nuxt
            ? 'nuxt-link' : 'router-link'
    }
  }
}
</script>
