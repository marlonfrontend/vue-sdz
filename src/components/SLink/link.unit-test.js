import { mount } from '@vue/test-utils'

import SLink from './Index.vue'

/* eslint-disable */
describe('SLink', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SLink, {
      propsData:{
        link: true,
        dark: true,
        to: 'https://seedz.ag'
      },
      slots:{
        default: 'Nome do Link'
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check class', () => {

    expect(cmp.classes()).toContain('c-link')

  })

  test('check body text', () => {

    const text = cmp.find('a')
    
    expect(text.exists()).toBe(true)
    expect(text.text()).toContain('Nome do Link')

  })

  test('check link', () => {

    const a = cmp.find('a')
    
    expect(a.exists()).toBe(true)
    expect(a.attributes().to).toContain('https://seedz.ag')

  })

})
