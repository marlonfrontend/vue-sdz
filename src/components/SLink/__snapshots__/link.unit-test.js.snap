// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SLink has the expected html structure 1`] = `
<a
  class="c-link"
  to="https://seedz.ag"
>
  Nome do Link
</a>
`;
